packages/snaps-execution-environments/src/common/validation.ts

```diff
@@ -308,6 +315,34 @@ export function assertIsOnUserInputRequestArguments(
   );
 }
 
+export const OnProtocolRequestArgumentsStruct = object({
+  scope: CaipChainIdStruct,
+  request: JsonRpcRequestStruct,
+}) as unknown as Struct<{ scope: CaipChainId; request: JsonRpcRequest }, null>;
```

Comment:
What's the reason for this?

Reply:
`The inferred type of 'OnProtocolRequestArgumentsStruct' references an inaccessible 'unique symbol' type. A type annotation is necessary.`

Reply:
And casting without unknown blows up due to the JSON types

Reply:
Is it not exported?

Reply:
guess not 🤷‍♂️ https://github.com/MetaMask/utils/blob/90837bad1d85277aa5b8f16fe40beb81e0a1e09d/src/json.ts#L94